pkg/interface/src/components/ErrorBoundary.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { Component } from 'react';
+import { RouteComponentProps, withRouter } from 'react-router-dom';
+
+
+class ErrorBoundary extends Component<
+  RouteComponentProps,
+  { error?: Error }
+> {
+  constructor(props) {
+    super(props);
+    this.state = { error: undefined };
+    const { history } = this.props;
+    history.listen((location, action) => {
+      if (this.state.error) {
+        this.setState({
+          error: undefined,
+        });
+      }
+    });
+  }
+
+  componentDidCatch(error) {
+    this.setState({ error });
+    return false;
+  }
+
+  render() {
+    if (this.state.error) {
+      return (
+        <div className="pa4">
+          <h1 className="mb4">Oh no</h1>
+          <p className="mb4">{this.state.error.message}</p>
+          <p>
+            <details>
+              <summary>Stack trace</summary>
+              <pre>{this.state.error.stack}</pre>
+            </details>
```

Comment:
You probably want to make it such that you can navigate out of the app from inside the error trace.